src/services/route-broadcaster.ts

```diff
@@ -481,24 +484,29 @@ export default class RouteBroadcaster {
 
     const currentBest = this.forwardingRoutingTable.get(prefix)
 
-    const currentNextHop = currentBest && currentBest.nextHop
+    const currentNextHop = currentBest && currentBest.route && currentBest.route.nextHop
     const newNextHop = route && route.nextHop
 
     if (currentNextHop !== newNextHop) {
-      if (route) {
-        this.forwardingRoutingTable.insert(prefix, route)
-      } else {
-        this.forwardingRoutingTable.delete(prefix)
-      }
-
       const epoch = this.forwardingRoutingTable.currentEpoch++
       const routeUpdate: RouteUpdate = {
         prefix,
         route,
         epoch
       }
+
+      if (route) {
+        this.forwardingRoutingTable.insert(prefix, routeUpdate)
+      } else {
+        this.forwardingRoutingTable.insert(prefix, routeUpdate)
```

Comment:
Should this be `.delete`?

Reply:
No, we need to keep track of the epoch of the last route change so we can later find it and remove it from the log easily. So we now want to keep an entry regardless of whether the route was added or removed. But, obviously, the `if` statement is superfluous now. D'oh. :)